### PR TITLE
prune: Handle --max-repack-size=0 as expected

### PR DESCRIPTION
**Pull request by MichaelEischer**
What does this PR change? What problem does it solve?
-----------------------------------------------------
Previously the flag was ignored and `--max-repack-size=1` had to be
used instead.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Mentioned in #1153.

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- ~~[ ] I have added tests for all code changes.~~
- ~~[ ] I have added documentation for relevant changes (in the manual).~~
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: changelog/unreleased/pull-3591 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Bugfix: Fix handling of `prune --max-repack-size=0`
+
+Restic ignored the `--max-repack-size` option when passing a value of 0. This
+has been fixed.
+
+As a workaround `--max-repack-size=1` can be used.
```
Comment (greatroar) [inline changelog/unreleased/pull-3591:6]:
```suggestion
As a workaround, `--max-repack-size=1` can be used with older versions of restic.
```

**Comment by greatroar** [comment]
LGTM.

File: changelog/unreleased/pull-3591 (hunk 1 above)
Comment (MichaelEischer) [reply changelog/unreleased/pull-3591]: Thanks for the suggestion.

File: cmd/restic/cmd_prune.go [hunk 2]
```diff
@@ -418,11 +419,7 @@ func prune(opts PruneOptions, gopts GlobalOptions, repo restic.Repository, usedB
 
 	for _, p := range repackCandidates {
 		reachedUnusedSizeAfter := (stats.size.unused-stats.size.remove-stats.size.repackrm < maxUnusedSizeAfter)
-
-		reachedRepackSize := false
-		if opts.MaxRepackBytes > 0 {
-			reachedRepackSize = stats.size.repack+p.unusedSize+p.usedSize > opts.MaxRepackBytes
-		}
+		reachedRepackSize := stats.size.repack+p.unusedSize+p.usedSize >= opts.MaxRepackBytes
```
Comment (MichaelEischer) [inline cmd/restic/cmd_prune.go:422]: I've change to comparison to `>=`. This ensures that `--max-repack-size=0` really repacks nothing at all.

**Review by rawtaz — APPROVED** [review]
Looks right to me.

**Pull request merged**